atc/api/accessor/handler.go

```diff
@@ -3,31 +3,38 @@ package accessor
 import (
 	"context"
 	"net/http"
+
+	"code.cloudfoundry.org/lager"
 )
 
 func NewHandler(
 	handler http.Handler,
 	accessFactory AccessFactory,
 	action string,
+	logger lager.Logger,
 ) http.Handler {
 	return accessorHandler{
 		handler:       handler,
 		accessFactory: accessFactory,
 		action:        action,
+		logger:        logger,
 	}
 }
 
 type accessorHandler struct {
 	handler       http.Handler
 	accessFactory AccessFactory
 	action        string
+	logger        lager.Logger
 }
 
 func (h accessorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-
 	acc := h.accessFactory.Create(r, h.action)
 	ctx := context.WithValue(r.Context(), "accessor", acc)
 
+	if acc, ok := acc.(Access); ok {
+		h.logger.Debug("audit", lager.Data{"action": r.URL.Path, "method": r.Method, "user": acc.UserName(), "parameters": r.URL.Query()})
```

Comment:
I think logging the full path and method might be too generic. [If you look at the way we set up routes](https://github.com/concourse/concourse/blob/master/atc/routes.go#L128), each one has a name (`GetJob`) and path params (e.g. `:team_name`). 

In the accessorHandler the name gets injected as `h.action` and the path params are available as `r.FormValues` (example [here](https://github.com/concourse/concourse/blob/master/atc/api/jobserver/get.go#L14)). They might also be available as query params but i'm not sure about that.

What if we logged just the `h.action` and path params along with the `UserName`. Most of our api requests don't supply many traditional query params, but we could potentially include those as well.

Thoughts?

Reply:
@pivotal-jwinters That sounds great to me. I'm grateful to get some feedback on it. as I was looking for something a bit more precise than just stating the full path and method.  

I had dived through the different attributes available, but must have missed those.  (especially as this is my first time looking through a lot of your codebase)